lib/ledger_sync/util/resources_builder.rb

```diff
@@ -41,12 +41,16 @@ def build_resource(external_id:, type:)
         resource_klass = LedgerSync.resources[type]
         raise "#{type} is an invalid resource type" if resource_klass.nil?
 
-        provided_references = resource_klass.references.select { |k, _v| current_data.key?(k) }
-
-        provided_references.each do |reference, reference_klass|
+        provided_references_one = resource_klass.references_one.select { |k, _v| current_data.key?(k) }
+        provided_references_one.each do |reference, reference_klass|
           current_data[reference] = resource_or_build(external_id: current_data[reference], type: reference_klass.resource_type)
         end
 
+        provided_references_many = resource_klass.references_many.select { |k, _v| current_data.key?(k) }
```

Comment:
Gosh this helper needs cleaned up.  Can refactor later, but welcome any input @mattgmarcus and @jozefvaclavik